RUFAS/biophysical/manure/handler/handler.py

```diff
@@ -0,0 +1,256 @@
+        cleaning_water_recycle_fraction: float,
+        use_parlor_flush: bool,
+    ):
+        super().__init__(name, is_housing_emissions_calculator=True)
+        self.manure_stream: ManureStream | None = None
+        self.fresh_water_volume_used_for_milking: float = 0.0
+        self.ammonia_emission: float = 0.0
+        self.handler_type = handler_type
+        self.cleaning_water_use_amount = cleaning_water_use_amount
+        self.minutes_per_cleaning = minutes_per_cleaning
+        self.cleanings_per_day = cleanings_per_day
+        self.cleaning_water_recycle_fraction = cleaning_water_recycle_fraction
+        self.use_parlor_flush = use_parlor_flush
+
+    def receive_manure(self, manure_stream: ManureStream) -> None:
+        """
+        Implements the basic checks for receiving manure stream.
+
+        Parameters
+        ----------
+        manure_stream : ManureStream
+            The ManureStream instance being checked for compatibility.
+
+        """
+        info_map = {"class": self.__class__.__name__, "function": self.receive_manure.__name__}
+        if not self.check_manure_stream_compatibility(manure_stream):
+            self._om.add_error(
+                "Invalid manure stream.",
+                "Received manure stream is not compatible with a handler type processor.",
+                info_map,
+            )
+            raise ValueError("ValueError: Invalid manure stream for handler processor.")
+
+    def process_manure(self, conditions: CurrentDayConditions, time: Time) -> dict[str, ManureStream]:
+        """
+        Executes the daily manure processing operations.
+
+        Parameters
+        ----------
+        conditions : CurrentDayConditions
+            Current weather and environmental conditions that manure is being processed in.
+        time : Time
+            Time instance containing the simulations temporal information.
+
+        Returns
+        -------
+        dict[str, ManureStream]
+            Mapping between classification of manure coming out of this processor to the ManureStream containing the
+            manure information. If the processor is a separator, the classifications are "solid" and "liquid". Otherwise
+            the only classification is "manure".
+
+        """
+        info_map: dict[str, Any] = {
+            "class": self.__class__.__name__,
+            "function": self.process_manure.__name__,
+            "prefix": self._prefix,
+            "simulation_day": time.simulation_day,
+            "handler_type": self.handler_type,
+        }
+        if self.manure_stream is None or self.manure_stream.pen_manure_data is None:
+            self._om.add_error(
+                "None type ManureStream.",
+                "The processed ManureStream or pen data of the manure stream is None type.",
+                info_map,
+            )
+            raise TypeError("TypeError: Handler tries to process 'NoneType' object ManureStream.")
```

Comment:
```suggestion
            raise TypeError("TypeError: Handler tried to process 'NoneType' object ManureStream or PenManureData.")
```

Reply:
Going to reiterate I think it would be helpful to have both possibilities for the type error here but also the TypeError in the actual message itself is redundant.
```suggestion
            raise TypeError("Handler tries to process 'NoneType' object ManureStream or PenManureData.")
```